### PR TITLE
8340532: C2: assert(is_OuterStripMinedLoop()) failed: invalid node class: IfTrue

### PR DESCRIPTION
**Pull request by rwestrel**
A `CountedLoopEnd` (that marks the end of a still existing
`CountedLoop`) is optimized out because a dominating identical
`CountedLoopEnd` (that no longer marks the end of an existing
`CountedLoop` but was left behind by previous loop opts) is
found. That causes the path out of `CountedLoopEnd` to become dead
including the `OuterStripMinedLoopEnd`. The `OuterStripMinedLoop`
looses its backedge as a consequence. The `CountedLoop` is still
marked as strip mined but the outer loop doesn't exist anymore.

The fix I propose for this corner case is to simply detect when that
happens (during igvn AFAICT) and clear the strip mined flag from the
`CountedLoop`.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8340532](https://bugs.openjdk.org/browse/JDK-8340532): C2: assert(is_OuterStripMinedLoop()) failed: invalid node class: IfTrue (**Bug** - P3)

### Reviewers
 * [Christian Hagedorn](https://openjdk.org/census#chagedorn) (@chhagedorn - **Reviewer**)
 * [Vladimir Kozlov](https://openjdk.org/census#kvn) (@vnkozlov - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/21956/head:pull/21956` \
`$ git checkout pull/21956`

Update a local copy of the PR: \
`$ git checkout pull/21956` \
`$ git pull https://git.openjdk.org/jdk.git pull/21956/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 21956`

View PR using the GUI difftool: \
`$ git pr show -t 21956`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/21956.diff">https://git.openjdk.org/jdk/pull/21956.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/21956#issuecomment-2462424494)
</details>

### CODE REVIEW

**Review by chhagedorn — APPROVED** [review]
Looks reasonable to me.

File: test/hotspot/jtreg/compiler/loopstripmining/TestIdenticalDominatingCLE.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test
+ * @bug 8340532
+ * @summary C2: assert(is_OuterStripMinedLoop()) failed: invalid node class: IfTrue
+ *
```
[the remark this answers is not in the document]
Comment (rwestrel) [reply test/hotspot/jtreg/compiler/loopstripmining/TestIdenticalDominatingCLE.java]: Right! Thanks.

**Review by vnkozlov — APPROVED** [review]
Good.

**Review by chhagedorn — APPROVED** [review]

**Comment by rwestrel** [comment]
@chhagedorn @vnkozlov thanks for the review

**Pull request closed without merging**